runtime/sema/errors.go

```diff
@@ -3004,11 +3005,21 @@ func (*InvalidAccessError) isSemanticError() {}
 func (*InvalidAccessError) IsUserError() {}
 
 func (e *InvalidAccessError) Error() string {
+	var possessedDescription string
+	if e.PossessedAccess != nil {
+		if e.PossessedAccess.Equal(UnauthorizedAccess) {
+			possessedDescription = ", but reference is unauthorized"
+		} else {
+			possessedDescription = fmt.Sprintf(", but reference only has %s access", e.PossessedAccess.Description())
+		}
+	}
+
 	return fmt.Sprintf(
-		"cannot access `%s`: %s has %s access",
+		"cannot access `%s`: %s requires %s access%s",
```

Comment:
Minor additional improvements: 
- In #2648, I was a bit confused by the "access" wording here: Functions have `access` declarations,  references have an (optional) `auth`orization. In https://github.com/onflow/cadence/pull/2648/files#diff-356e61512ff8ef6d2b560c92b219ee43ad912048284bce6bf056b7bc0725cc6dR3039 I reworded "access" here to "authorization"
- Maybe put the authorization ("access") in quotes, because it may be e.g. a conjunction/disjunctions of multiple entitlements
```suggestion
		"cannot access `%s`: %s requires `%s` authorization%s",
```